shared/js/contacts/import/utilities/overlay.js

```diff
@@ -76,119 +179,104 @@ var utils = window.utils || {};
-    progressElement.setAttribute('value', '0');
+      overlay.addEventListener('animationend',
+        function ov_showWhenFinished(ev) {
+          overlay.removeEventListener('animationend', ov_showWhenFinished);
+          _show(messageId);
+        }
+      );
+    }).catch(function() {
+      deleteOverlay();
+    });
   }
 
-  function setClass(clazzName) {
-    switch (clazzName) {
-      case 'spinner':
-        progressElement.classList.remove('pack-activity');
-        statusContainer.classList.add('loading-icon');
-        progressElement.removeAttribute('max');
-        progressElement.removeAttribute('value');
-      break;
-      case 'activityBar':
-      case 'progressActivity':
-        progressElement.classList.add('pack-activity');
-        setAsProgress();
-      break;
-      case 'progressBar':
-        progressElement.classList.remove('pack-activity');
-        setAsProgress();
-      break;
-    }
+  function hideMenu() {
+    menu.classList.remove('showed');
   }
 
-  utils.overlay._hide = function ov__hide() {
-    if (!utils.overlay.isShown) {
+  function _hide() {
+    if (!isShown) {
       return;
     }
     overlay.classList.remove('fade-in');
     overlay.classList.add('fade-out');
-    utils.overlay.isAnimationPlaying = true;
+    isAnimationPlaying = true;
     // Custom event that can be used to apply (screen reader) visibility
     // changes.
     window.dispatchEvent(new CustomEvent('loadingoverlayhiding'));
     overlay.addEventListener('animationend', function ov_onFadeOut(ev) {
-      utils.overlay.isAnimationPlaying = false;
+      isAnimationPlaying = false;
       overlay.removeEventListener('animationend', ov_onFadeOut);
       progressActivity.classList.add('hide');
       overlay.classList.add('no-opacity');
       overlay.classList.add('hide');
-      utils.overlay.isShown = false;
+      isShown = false;
+      window.dispatchEvent(new CustomEvent('overlayhidden'));
+      deleteOverlay();
     });
-  };
-  utils.overlay.hide = function ov_hide() {
-    if (!utils.overlay.isAnimationPlaying) {
-      utils.overlay._hide();
+  }
+
+  function hide() {
+    // Avoid showing the overlay if it's not necessary
+    if (isLoading) {
+      _reject();
+      return;
+    }
+
+    if (!isAnimationPlaying) {
+      _hide();
       return;
     }
     overlay.addEventListener('animationend',
       function ov_hideWhenFinished(ev) {
         overlay.removeEventListener('animationend', ov_hideWhenFinished);
-        utils.overlay._hide();
+        _hide();
       }
     );
+  }
+
+  exports.Overlay = {
+    'show': show,
+    'hide': hide,
+    'hideMenu': hideMenu,
+    'update': update,
+    'setHeaderMsg': updateHeaderMsg,
+    'setProgressType': setProgressType,
+    'setTotal': setTotal,
+    set oncancel(cancelCb) {
+      if (typeof cancelCb === 'function') {
+        _cancelCB = cancelCb;
+      } else {
+        _cancelCB = function foo() {};
+      }
+    }
```

Comment:
I dont like the interface exposed here, I think is quite complicated to understand. Please simplify this (I know you are trying to keep the old methods exposed :) ).